python/cudf/cudf/tests/test_orc.py

```diff
@@ -1880,3 +1880,13 @@ def test_orc_reader_apache_negative_timestamp(datadir):
     gdf = cudf.read_orc(path)
 
     assert_eq(pdf, gdf)
+
+
+def test_statistics_string_sum():
+    strings = ["a string", "another string!"]
+    buff = BytesIO()
+    df = cudf.DataFrame({"str": strings})
+    df.to_orc(buff)
+
+    file_stats, stripe_stats = cudf.io.orc.read_orc_statistics([buff])
+    assert_eq(file_stats[0]["str"].get("sum"), sum(len(s) for s in strings))
```

Comment:
Pardon me, but wouldn't this test pass even without the `uint64_t` -> `int64_t` change?

Reply:
I wondered the same thing, I suspect it, check the data type and not just the value.

Reply:
The output value is not the same without the fix because ORC uses zigzag encoding for signed integers so messing up the type changes the read value. So, even sums are read as sum/2, and odd ones as -(sum+1)/2.
Apologies, should have included this in the PR description.